includes/abandoned-signups.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+/**
+ * Set up a user taxonomy to track users who were created during the PMPro
+ * checkout process but did not complete the checkout.
+ *
+ * @since TBD
+ */
+function pmpro_abandoned_signups_taxonomy() {
+	// Register the taxonomy.
+	register_taxonomy(
+		'pmpro_abandoned_signup',
+		'user',
+		array(
+			'public'            => false,
+			'default_term'      => array(
+				'slug' => 'abandoned-signup',
+			),
+		)
+	);
+}
+add_action( 'init', 'pmpro_abandoned_signups_taxonomy', 10 );
+
+/**
+ * Add the abandoned signup taxonomy to the user object when they are created during
+ * the PMPro checkout process.
+ *
+ * @since TBD
+ *
+ * @param int $user_id The user ID.
+ */
+function pmpro_set_abandoned_signup_taxonomy( $user_id ) {
+	// Bail if the user ID is empty.
+	if ( empty( $user_id ) ) {
+		return;
+	}
+
+	// Add the abandoned signup taxonomy to the user.
+	var_dump( wp_set_object_terms( $user_id, 'abandoned-signup', 'pmpro_abandoned_signup' ) );
+}
+add_action( 'pmpro_checkout_before_user_auth', 'pmpro_set_abandoned_signup_taxonomy', 10 );
+
+/**
+ * After a checkout is complete, remove the abandoned signup taxonomy from the user.
+ *
+ * @param int $user_id The user ID.
+ */
+function pmpro_remove_abandoned_signup_taxonomy( $user_id ) {
+	// Bail if the user ID is empty.
+	if ( empty( $user_id ) ) {
+		return;
+	}
+
+	// Remove the abandoned signup taxonomy from the user.
+	wp_remove_object_terms( $user_id, 'abandoned-signup', 'pmpro_abandoned_signup' );
+}
+add_action( 'pmpro_after_checkout', 'pmpro_remove_abandoned_signup_taxonomy' );
+add_action( 'deleted_user', 'pmpro_remove_abandoned_signup_taxonomy' );
+
+/**
+ * Filter the views on the Users page to include a view for abandoned signups.
+ *
+ * @since TBD
+ *
+ * @param array $views The views on the Users page.
+ */
+function pmpro_add_users_table_view_abandoned_signups( $views ) {
```

Comment:
If we have 3 statuses now, we probably want to at least allow filtering by pending as well. I don't think filtering by completed is as useful, but maybe people would expect it.

Reply:
No longer relevant if we go back to 1 or 2 statuses.